### PR TITLE
fix: [BB] 90 fix pg module password error

### PR DESCRIPTION
**Pull request by GDamaso**
# Description

Fixed a bug that was caused by jest testing in the backend. There were no environment vars being set on local environment, where the tests run from.
Thanks to @LocalNewsTV for helping me figure this out.

Now Jest will use dotenv to set things up and work as expected.

There is a setup ts file setting the test endpoint and the dotenv.config, exporting a configured test request for the endpoint tests to use.

I also took the time to do some clean up on docs, docker-compose and commenting. 

## Type of changes

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

I was thrown off initially because the error pointed to an issue with postgres npm module not getting a string for the db password. 

image

Now when running `npm run test` it calls for a script `"test": "jest --setupFiles dotenv/config",` with the correct result:

image

**Please be mindful that this is not a PR for the backend testing implementation, but a bug fix that arose during the OpenShift first test deploy and the Jest backend implementation.*

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-91-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-91-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-91-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-91-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

File: backend/src/test/developers.test.ts
[the remark this answers is not in the document]
Comment (LocalNewsTV) [reply backend/src/test/developers.test.ts:1]: Optional but handy tool to keep in the back pocket https://www.freecodecamp.org/news/how-to-check-if-an-object-has-a-key-in-javascript/

**Review by LocalNewsTV — COMMENTED** [review]
Technically correct *(the best kind of correct)*
I encourage you to spend a little more time tightening your test on the developer endpoint

Also: 
- Write a Wiki page about testing with Jest! You went through trial and error to get things up and running, had to request outsider help KEEP A RECORD! This benefits you, your team, and others! Imagine you were leading a Capstone team, you don't want to use things **my team** did as a reference, you need your own things too :)

- This doesn't have to live in your Wiki, but I encourage your team to keep some sort of log about all the lessons you're learning while doing this Capstone. In demonstrating ones own capability, having a record of all the work and learning you did over a short period of time can come in handy! 
    - Save the small tutorials for the Wiki though, that's good as gold, and what a way to showcase to potential employers!

File: backend/README.md [hunk 1]
```diff
@@ -8,7 +8,6 @@ Documentation is accessible through Swagger.
 
 ## Accessing the API Documentation
 
-
 http://localhost:3000/api/api-docs
```
Comment (LocalNewsTV) [inline backend/README.md:11]: Once up and running this should be a link to your OpenShift realm, if there is a permanent one 😏

File: backend/src/test/developers.test.ts
[the remark this answers is not in the document]
Comment (GDamaso) [reply backend/src/test/developers.test.ts:1]: Thanks @LocalNewsTV, I'll use these comments for improving my tests. 
This branch is mostly for working on a fix for the password bug that showed up during its implementation, but the comments hold true and will help me in making a better PR for branch [`75-develop-automated-tests-for-api`](https://github.com/bcgov/nr-sustainment-capstone-2024/tree/75-develop-automated-tests-for-api)

**Comment by LocalNewsTV** [comment]
# Why to check beyond a status 200

image

File: backend/src/test/developers.test.ts
[the remark this answers is not in the document]
Comment (LocalNewsTV) [reply backend/src/test/developers.test.ts:1]: @GDamaso You are correct in wanting to implement `one ticket, one branch`, but your branch is now carrying 12 files worth of changes containing:

1. Wide documentation updates
2. Two Test suites
3. Package updates
4. The bugfix

So it's quite out of scope for the work of just fixing a bug
At this point its easier to just add the one extra test and tag it to two issues.
It's not a problem if you don't make it a habit :) Sometimes its easy to get carried away, especially this early in development.

[the remark this answers is not in the document]
Comment (LocalNewsTV) [reply backend/src/test/developers.test.ts:1]: I definitely had some PR's containing 2-3 tickets worth of work during Capstone😏

**Review by LocalNewsTV — APPROVED** [review]

**Comment by LocalNewsTV** [comment]
Absolute overkill, but you're 100% right, the goal is learning, documenting learning. And skilling up as a team :) if this serves as a starting example it'll help you as you progress

**Comment by GDamaso** [comment]
Wiki with the bug fix explanation:
https://github.com/bcgov/nr-sustainment-capstone-2024/wiki/Lessons-Learned

**Pull request merged**